MonstieWash/Assets/Scripts/UI/NagivationUI.cs

```diff
@@ -0,0 +1,42 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+/// <summary>
+/// Generates nav UI. Finds scene camera, sets it to Canvas render camera.
+/// </summary>
+public class NavigationUI : MonoBehaviour
+{
+    private List<GameScene> gameScenes;
+    [SerializeField] private GameObject buttonTile;
+    [SerializeField] private GameObject navPanel;
+
+    [SerializeField] private float tilePadding = 1.25f; //Seems like a nice number. Means that the UI doesn't overlap with the bag on Mimic.
+
+    void Awake()
+    {
+        Canvas canvas = GetComponent<Canvas>();
+        Camera camera = FindFirstObjectByType<Camera>();
+
+        canvas.worldCamera = camera;
+
+        gameScenes = GameSceneManager.Instance.GetLevelScenes();
+
+        InstantiateUI();
+    }
+
+    private void InstantiateUI ()
+    {
+        //Resize Nav Panel to fit all tiles.
+        RectTransform npSize = navPanel.GetComponent<RectTransform>();
+        npSize.sizeDelta = new Vector2(tilePadding, gameScenes.Count * tilePadding);
```

Comment:
### 🗒️ **Convention** | _This is for maintaining our coding conventions._

Teeny tiny lil fix to use `var` lol

```suggestion
        var npSize = navPanel.GetComponent<RectTransform>();
        npSize.sizeDelta = new Vector2(tilePadding, gameScenes.Count * tilePadding);
```